File: data/readable_format/fortran/file2210.f90
program test_ynorm
    implicit none

    integer, parameter :: maxsiz = 10
    double precision :: observ(maxsiz), rnorm
    integer :: jstrt, jfins
    integer :: i

    ! Initialize the observ array and other variables
    rnorm = 2.0d0
    jstrt = 2
    jfins = 5

    ! Populate the observ array with initial values
    do i = 1, maxsiz
        observ(i) = dble(i)
    end do

    ! Call the ynorm subroutine
    call ynorm(maxsiz, jstrt, jfins, observ, rnorm)

    ! Print the results to verify the computation
    print *, 'Observ array after calling ynorm:'
    do i = 1, maxsiz
        print *, observ(i)
    end do

end program test_ynorm

subroutine ynorm(maxsiz, jstrt, jfins, observ, rnorm)
    implicit none

    integer maxsiz, jstrt, jfins
    double precision observ(maxsiz), rnorm
    integer i_res

    do 515 i_res = jstrt, jfins
        observ(i_res) = rnorm * observ(i_res)
515 continue
end subroutine ynorm

File: data/readable_format/cpp/file2210.cpp
#include <iostream>
#include <vector>

void ynorm(int maxsiz, int jstrt, int jfins, std::vector<double> &observ, double rnorm) {
    // Adjusting jstrt and jfins for 0-based indexing
    jstrt -= 1;
    jfins -= 1;
    for (int i_res = jstrt; i_res <= jfins; ++i_res) {
        observ[i_res] = rnorm * observ[i_res];
    }
}

int main() {
    const int maxsiz = 10;
    std::vector<double> observ(maxsiz);
    double rnorm = 2.0;
    int jstrt = 2; // Adjusted in the ynorm function to match Fortran 1-based indexing
    int jfins = 5; // Adjusted in the ynorm function to match Fortran 1-based indexing

    // Populate the observ array with initial values
    for (int i = 0; i < maxsiz; ++i) {
        observ[i] = static_cast<double>(i + 1);
    }

    // Call the ynorm function
    ynorm(maxsiz, jstrt, jfins, observ, rnorm);

    // Print the results to verify the computation
    std::cout << "Observ array after calling ynorm:" << std::endl;
    for (int i = 0; i < maxsiz; ++i) {
        std::cout << observ[i] << " ";
    }
    std::cout << std::endl;

    return 0;
}